### DISCUSSION TITLE
Significant time difference when performing 'cue eval' with multiple files

### DISCUSSION THREAD

**Discussion by lemigu** [cue-lang/cue#1756] [Q&A] [answered]
I am using CUE in the context of validating GitHub workflows. I have multiple `.cue` files, one for each individual workflow, and one containing all global definitions and constraints, as follows:

```
working-directory
  - workflowA.cue
  - workflowB.cue
  - workflowC.cue
  - constraints.cue
```

When calling `cue eval` to check each workflow against the constraints, if I make one call per workflow file:

```
cue eval constraints.cue workflowA.cue
cue eval constraints.cue workflowB.cue
cue eval constraints.cue workflowC.cue
```

each call to `cue eval` takes between 1 and 5 seconds to complete.

However, when making a single call to check all workflows simultaneously:

```
cue eval constraints.cue workflowA.cue workflowB.cue workflowC.cue
```

it takes up to 20minutes to complete.

Furthermore, if I join the contents of `workflowA.cue`, `workflowB.cue` and `workflowC.cue` in a single `workflows.cue` file, it still only takes 6 to 15 seconds to complete.

Is there an explanation as to what may be causing this discrepancy in execution times?

**Comment by mvdan** [comment, marked as answer]
Could possibly be https://github.com/cue-lang/cue/issues/1742; there, @myitcv reports that duplicating schemas as part of a single `eval` invocation makes the runtime jump significantly.

**Reply by lemigu** [reply]
In my particular scenario, there were several duplicate schemas, but the ones causing the significant impact were:

`#job:  ((github.#Workflow & {}).jobs & {x: _}).x`
`#step: ((#job & {steps:                 _}).steps & [_])[0]`

Thanks!
